File: 1935. Maximum Number of Words You Can Type/solution.java
class Solution {
  public int canBeTypedWords(String text, String brokenLetters) {
    Set<Character> letterSet = new HashSet<>();
    for (int index = 0; index < brokenLetters.length(); index++){
      letterSet.add(brokenLetters.charAt(index));
    }
    int count = 0;
    boolean word = false;
    for (int index = 0; index < text.length(); index++){
      Character letter = text.charAt(index);
      if (word && letter != ' '){
        continue;
      }
      if (letterSet.contains(letter)){
        word = true;
      }
      if (letter == ' ' || index == text.length() - 1){
        if (word == false){
          count += 1;
        }
        word = false;
      }
    }

    return count;
  }
}

File: 1935. Maximum Number of Words You Can Type/solution.js
/**
 * @param {string} text
 * @param {string} brokenLetters
 * @return {number}
 */
 var canBeTypedWords = function(text, brokenLetters) {
  let letterSet = new Set(brokenLetters);
  let count = 0;
  let word = false;
  for (let index = 0; index < text.length; index++){
    let letter = text[index];
    if (word && letter != " "){
      continue;
    }
    if (letterSet.has(letter)){
      word = true;
    }
    if (letter == " " || index == text.length - 1){
      if (word == false){
        count += 1;
      }
      word = false;
    }
  }

  return count;
};